lib/websocket.js

```diff
@@ -466,10 +466,33 @@ function initAsClient(websocket, address, protocols, options) {
   }
 
   let parsedUrl;
+  let isWinPipe;
 
   if (address instanceof URL) {
     parsedUrl = address;
     websocket.url = address.href;
+  } else if ((isWinPipe = /^ws\+unix:\/\/\\\\[.?]\\pipe\\/.test(address))) {
```

Comment:
I'm not happy to skip the URL parser which is why there is no IPC support on Windows. This is not a valid URL.